src/starknet_runner/runner.rs

```diff
@@ -377,4 +377,114 @@ mod tests {
+        let class_hash = [1; 32];
+        let contract_state = ContractState::create(class_hash, 3.into(), HashMap::new());
+
+        contract_class_cache.insert(class_hash, contract_class);
+        let mut state_reader = InMemoryStateReader::new(ffc, contract_class_storage);
+        state_reader
+            .contract_states
+            .insert(address.clone(), contract_state);
+
+        //* ---------------------------------------
+        //*    Create state with previous data
+        //* ---------------------------------------
+
+        let mut state = CachedState::new(state_reader, Some(contract_class_cache));
+
+        //* ------------------------------------
+        //*    Create execution entry point
+        //* ------------------------------------
+
+        let calldata = [].to_vec();
+        let caller_address = Address(0000.into());
+        let entry_point_type = EntryPointType::External;
+
+        let exec_entry_point = ExecutionEntryPoint::new(
+            address,
+            calldata.clone(),
+            storage_entrypoint_selector.clone(),
+            caller_address,
+            entry_point_type,
+            Some(CallType::Delegate),
+            Some(class_hash),
+        );
+
+        //* --------------------
+        //*   Execute contract
+        //* ---------------------
+        let general_config = StarknetGeneralConfig::default();
+        let tx_execution_context = TransactionExecutionContext::new(
+            Address(0.into()),
+            Felt::zero(),
+            Vec::new(),
+            0,
+            10.into(),
+            general_config.invoke_tx_max_n_steps,
+            TRANSACTION_VERSION,
+        );
+        let mut resources_manager = ExecutionResourcesManager::default();
+
+        let expected_key = calculate_sn_keccak("_counter".as_bytes());
+
+        let mut expected_accesed_storage_keys = HashSet::new();
+        expected_accesed_storage_keys.insert(expected_key);
+
+        let expected_call_info = CallInfo {
+            caller_address: Address(0.into()),
+            call_type: Some(CallType::Delegate),
+            contract_address: Address(1111.into()),
+            entry_point_selector: Some(storage_entrypoint_selector),
+            entry_point_type: Some(EntryPointType::External),
+            calldata,
+            retdata: [1.into(), 42.into()].to_vec(),
+            execution_resources: ExecutionResources::default(),
+            class_hash: Some(class_hash),
+            storage_read_values: vec![42.into()],
+            accesed_storage_keys: expected_accesed_storage_keys,
+            ..Default::default()
+        };
+
+        assert_eq!(
+            exec_entry_point
+                .execute(
+                    &mut state,
+                    &general_config,
+                    &mut resources_manager,
+                    &tx_execution_context
+                )
+                .unwrap(),
+            expected_call_info
+        );
+    }
```

Comment:
Here we need to add other check that test that the State is modified. 
I think that first we need to check that the Contract storage in the `StateCache` was modified.

Reply:
Maybe also that is written in the InMemoryStateReader? @juanbono